### PR TITLE
Bug 1392664 - Add SentryIntegration API for saving tags/attributes to…

### PR DESCRIPTION
**Pull request by justindarc**
… all logs/crash reports

### CODE REVIEW

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
BrowserDB.swift | 208 | Explicitly calling .init() should be avoided. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Shared/SentryIntegration.swift [hunk 1]
```diff
@@ -96,11 +102,15 @@ public class SentryIntegration {
...
-            event.tags = ["event": tag]
+            event.tags = ["tag": tag]
 
             Client.shared?.appendStacktrace(to: event)
             event.debugMeta = nil
             Client.shared?.send(event: event, completion: completion)
         }
     }
+
+    public func setAttributes(_ attributes: [String : Any]) {
```
Comment (st3fan) [inline Shared/SentryIntegration.swift]: Let's call this `addAttributes` if it merges?

File: Shared/SentryIntegration.swift [hunk 2]
```diff
@@ -79,7 +85,7 @@ public class SentryIntegration {
 
         let event = Event(level: severity)
         event.message = message
-        event.tags = ["event": tag]
```
Comment (st3fan) [inline Shared/SentryIntegration.swift]: Is this going to give us trouble when mixing existing and new reports? In terms of search queries? I understand it is incorrect but maybe leave it?

File: Storage/SQL/BrowserDB.swift [hunk 3]
```diff
@@ -165,6 +171,9 @@ open class BrowserDB {
                 }
             }
 
+            log.info("Attempting to update schema from version \(currentVersion) to \(schema.version).")
+            SentryIntegration.shared.sendWithStacktrace(message: "Attempting to update schema from version \(currentVersion) to \(schema.version).", tag: "BrowserDB", severity: .info)
```
Comment (st3fan) [inline Storage/SQL/BrowserDB.swift]: Do we need the stack trace here?

File: Shared/SentryIntegration.swift (hunk 1 above)
Comment (justindarc) [inline Shared/SentryIntegration.swift]: It would also override existing values which is why I named it that. It doesn't matter much to me though.

**Pull request merged**